xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -138,23 +155,36 @@ bool CWinSystemGbm::ResizeWindow(int newWidth, int newHeight, int newLeft, int n
 
 bool CWinSystemGbm::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, bool blankOtherDisplays)
 {
-  if(!CDRMUtils::SetMode(res))
+  if(!m_DRM->SetMode(res))
   {
     CLog::Log(LOGERROR, "CWinSystemGbm::%s - failed to set DRM mode", __FUNCTION__);
     return false;
   }
 
-  return m_DRM.SetVideoMode(res);
+  struct gbm_bo *bo = m_GBM->LockFrontBuffer();
+
+  if(!m_DRM->SetVideoMode(res, bo))
```

Comment:
`ReleaseBuffer` is never called if `SetVideoMode` fails, recommend changing to something like:
```
struct gbm_bo *bo = m_GBM->LockFrontBuffer();
bool result = m_DRM->SetVideoMode(res, bo);
m_GBM->ReleaseBuffer();
return result;
```